### ISSUE TITLE
Async daemon - Shard never reaches most up to date sequence number

### ISSUE THREAD

**Issue by julienperignon** [JasperFx/marten#1858]
Hi there,

The Shard for one of the two async projections I have set up never reaches the latest sequence number when multiple projections are configured and events are fired on both.

I encountered this issue while writing AggregateProjections. The projection would just stop updating even though events are getting fired. Once the app is restarted the daemon seems to be catching up.

I will send a pull request shortly with a reproducible unit test.

Cheers :)

**Comment by julienperignon** [comment]
Info from the unit test logger : 

```Message: 
    System.TimeoutException : Shard TruckView:All did not reach sequence number 4 in the time allowed

  Stack Trace: 
    build_aggregate_multiple_projections.bug_repro() line 126
    --- End of stack trace from previous location ---

  Standard Output: 
    IProjection/Information: Started HighWaterAgent
    IProjection/Debug: Received ShardName: HighWaterMark, Sequence: 0, Action: Started
    IProjection/Information: Starting projection agent for 'Identity: CarView:All'
    IProjection/Information: Projection agent for 'Identity: CarView:All' has started from sequence 0 and a high water mark of 0
    IProjection/Debug: Received ShardName: CarView:All, Sequence: 0, Action: Started
    IProjection/Information: Starting projection agent for 'Identity: TruckView:All'
    IProjection/Information: Projection agent for 'Identity: TruckView:All' has started from sequence 0 and a high water mark of 0
    IProjection/Debug: Received ShardName: TruckView:All, Sequence: 0, Action: Started
    IProjection/Debug: High Water mark detected at 1
    IProjection/Debug: Received ShardName: HighWaterMark, Sequence: 1, Action: Updated
    IProjection/Debug: Projection Shard 'Identity: CarView:All' received high water mark at 1
    IProjection/Debug: Projection Shard 'Identity: TruckView:All' received high water mark at 1
    IProjection/Debug: Enqueued processing of Event range of 'Identity: TruckView:All', 0 to 1
    IProjection/Debug: Enqueued processing of Event range of 'Identity: CarView:All', 0 to 1
    IProjection/Debug: Loaded events for Event range of 'Identity: TruckView:All', 0 to 1
    IProjection/Debug: Loaded events for Event range of 'Identity: CarView:All', 0 to 1
    IProjection/Debug: Shard 'Identity: CarView:All':Starting to group Event range of 'Identity: CarView:All', 0 to 1
    IProjection/Debug: Shard 'Identity: CarView:All': successfully grouped Event range of 'Identity: CarView:All', 0 to 1
    IProjection/Debug: Shard 'Identity: CarView:All': Starting to process events for Aggregate for Event range of 'Identity: CarView:All', 0 to 1, 1 slices
    IProjection/Information: Shard 'Identity: CarView:All': Executed updates for Event range of 'Identity: CarView:All', 0 to 1
    IProjection/Debug: Received ShardName: CarView:All, Sequence: 1, Action: Updated
    IProjection/Debug: Shard 'Identity: CarView:All': Configured batch Aggregate for Event range of 'Identity: CarView:All', 0 to 1, 1 slices
    IProjection/Debug: High Water mark detected at 4
    IProjection/Debug: Received ShardName: HighWaterMark, Sequence: 4, Action: Updated
    IProjection/Debug: Projection Shard 'Identity: CarView:All' received high water mark at 4
    IProjection/Debug: Projection Shard 'Identity: TruckView:All' received high water mark at 4
    IProjection/Debug: Enqueued processing of Event range of 'Identity: CarView:All', 1 to 4
    IProjection/Debug: Enqueued processing of Event range of 'Identity: TruckView:All', 1 to 4
    IProjection/Debug: Loaded events for Event range of 'Identity: CarView:All', 1 to 4
    IProjection/Debug: Shard 'Identity: CarView:All':Starting to group Event range of 'Identity: CarView:All', 1 to 4
    IProjection/Debug: Shard 'Identity: CarView:All': successfully grouped Event range of 'Identity: CarView:All', 1 to 4
    IProjection/Debug: Shard 'Identity: CarView:All': Starting to process events for Aggregate for Event range of 'Identity: CarView:All', 1 to 4, 1 slices
    IProjection/Debug: Loaded events for Event range of 'Identity: TruckView:All', 1 to 4
    IProjection/Information: Shard 'Identity: CarView:All': Executed updates for Event range of 'Identity: CarView:All', 1 to 4
    IProjection/Debug: Shard 'Identity: CarView:All': Configured batch Aggregate for Event range of 'Identity: CarView:All', 1 to 4, 1 slices
    IProjection/Debug: Received ShardName: CarView:All, Sequence: 4, Action: Updated
```

**Comment by julienperignon** [comment]
fixed in 1d271a821e6f39bfd2571d77776502dca8bf5fd3 ! thanks :)
